Wikipedia/Code/WKWebView+EditSelectionJavascript.swift

```diff
@@ -0,0 +1,63 @@
+
+import WebKit
+
+extension WKWebView {
+    private func selectedTextEditInfo(from dictionary: Dictionary<String, Any>) -> SelectedTextEditInfo? {
+        guard
+            let selectedAndAdjacentTextDict = dictionary["selectedAndAdjacentText"] as? Dictionary<String, Any>,
+            let selectedText = selectedAndAdjacentTextDict["selectedText"] as? String,
+            let textBeforeSelectedText = selectedAndAdjacentTextDict["textBeforeSelectedText"] as? String,
+            let textAfterSelectedText = selectedAndAdjacentTextDict["textAfterSelectedText"] as? String,
+            let isSelectedTextInTitleDescription = dictionary["isSelectedTextInTitleDescription"] as? Bool,
+            let sectionID = dictionary["sectionID"] as? Int
+            else {
+                DDLogError("Error converting dictionary to SelectedTextEditInfo")
+                return nil
+        }
+        let selectedAndAdjacentText = SelectedAndAdjacentText(selectedText: selectedText, textAfterSelectedText: textAfterSelectedText, textBeforeSelectedText: textBeforeSelectedText)
+        return SelectedTextEditInfo(selectedAndAdjacentText: selectedAndAdjacentText, isSelectedTextInTitleDescription: isSelectedTextInTitleDescription, sectionID: sectionID)
+    }
+    
+    @objc func wmf_getSelectedTextEditInfo(completionHandler: ((SelectedTextEditInfo?, Error?) -> Void)? = nil) {
+        evaluateJavaScript("window.wmf.editTextSelection.getSelectedTextEditInfo()") { (result, error) in
+            guard let error = error else {
+                guard let completionHandler = completionHandler else {
+                    return
+                }
+                guard
+                    let resultDict = result as? Dictionary<String, Any>,
+                    let selectedTextEditInfo = self.selectedTextEditInfo(from: resultDict)
+                else {
+                    DDLogError("Error handling 'getSelectedTextEditInfo()' dictionary response")
+                    return
+                }
+                
+                completionHandler(selectedTextEditInfo, nil)
+                return
+            }
+            DDLogError("Error when evaluating javascript on fetch and transform: \(error)")
+        }
+    }
+}
+
+@objcMembers class SelectedAndAdjacentText: NSObject  {
```

Comment:
does this need to be exposed to objc and inherit from `NSObject`? it's only used in Swift

Reply:
good catch! fixed!